src/util.c

```diff
@@ -414,10 +385,18 @@ util_init (void)
 void*
 objid (const void *ptr)
 {
-    g_assert (ptr_mask != NULL);
+    GChecksum *checksum;
+    uintptr_t buf [64 / sizeof (uintptr_t)] = { 0, };
```

Comment:
if sizeof(uintptr_t) is ever > 64, this fails.... uintptr_t buf[0] is valid in GCC, but is discouraged:
```
Declaring zero-length arrays in other contexts, including as interior members of structure objects or as non-member objects, is discouraged. Accessing elements of zero-length arrays declared in such contexts is undefined and may be diagnosed.
```
You would want something like:
```C
(sizeof(uintptr_t) * 8) / sizeof(uintptr_t))
```